hybrid/pkg/agent/agent.go

```diff
@@ -0,0 +1,169 @@
+		result := elem.MethodByName("AidAttestation").Call([]reflect.Value{reflect.ValueOf(newInterceptor)})
+		err := result[0].Interface()
+		if err != nil {
+			return status.Errorf(codes.Internal, "An error ocurred when during AidAttestation.")
+		}
+	}
+
+	combinedMessage := common.PluginMessageList{}
+	for i := 0; i < len(interceptors); i++ {
+		combinedMessage.Messages = append(combinedMessage.Messages, interceptors[i].GetMessage())
+	}
+
+	return p.interceptor.SendCombined(combinedMessage)
+}
+
+func (p *HybridPluginAgent) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	pluginData, _ := p.decodeStringAndTransformToAstNode(req.HclConfiguration)
+
+	pluginNames, pluginsData := p.parseReceivedData(pluginData)
+
+	p.pluginList, p.initStatus = p.initPlugins(pluginNames)
+
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return nil, p.initStatus
+	}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		req.HclConfiguration = pluginsData[p.pluginList[i].PluginName]
+
+		result := elem.MethodByName("Configure").Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(req)})
+		err := result[1]
+
+		if !err.IsNil() {
+			return &configv1.ConfigureResponse{}, status.Errorf(codes.Internal, "Error configuring one of the supplied plugins.")
+		}
+	}
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+func (p *HybridPluginAgent) decodeStringAndTransformToAstNode(hclData string) (common.Generics, error) {
+	var genericData common.GenericPluginSuper
+	if err := hcl.Decode(&genericData, hclData); err != nil {
+		return nil, err
+	}
+
+	var data bytes.Buffer
+	printer.DefaultConfig.Fprint(&data, genericData.Plugins)
+
+	var astNodeData common.Generics
+
+	if err := hcl.Decode(&astNodeData, data.String()); err != nil {
+		return nil, err
+	}
+
+	return astNodeData, nil
+}
+
+func (p *HybridPluginAgent) parseReceivedData(data common.Generics) ([]string, map[string]string) {
+	str := map[string]string{}
+	plugins := []string{}
+	for key := range data {
+		var data_ bytes.Buffer
+		printer.DefaultConfig.Fprint(&data_, data[key])
+		str[key] = strings.Replace(strings.Replace(data_.String(), "{", "", -1), "}", "", -1)
+		plugins = append(plugins, key)
+	}
+	return plugins, str
+}
+
+func (p *HybridPluginAgent) initPlugins(pluginList []string) ([]common.Types, error) {
+	attestors := make([]common.Types, 0)
+
+	for i := 0; i < len(pluginList); i++ {
+		var plugin common.Types
+		switch pluginList[i] {
+		case "aws_iid":
+			plugin.PluginName = "aws_iid"
+			plugin.Plugin = awsiid.New()
+		case "k8s_psat":
+			plugin.PluginName = "k8s_psat"
+			plugin.Plugin = k8spsat.New()
+		case "azure_msi":
+			plugin.PluginName = "azure_msi"
+			plugin.Plugin = azuremsi.New()
+		case "gcp_iit":
+			plugin.PluginName = "gcp_iit"
+			plugin.Plugin = gcpiit.New()
+		default:
+			plugin.PluginName = ""
+			plugin.Plugin = nil
+		}
+
+		attestors = append(attestors, plugin)
+	}
```

Comment:
since this is a fixed list, I suggest we add the list of supported node attestors in the README, as new introduced built-in plugins will need a new version of this attestor with the new case clause.

Reply:
Agreed. Already added to README